src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,396 @@
+}
+
+void CubeMap::prepareToDraw(int cubeSideIndex) {
+  Magnum::GL::CubeMapCoordinate cubeMapCoord =
+      convertFaceIndexToCubeMapCoordinate(cubeSideIndex);
+
+  if (flags_ & Flag::ColorTexture) {
+    frameBuffer_.attachCubeMapTexture(
+        colorAttachment, *textures_[TextureType::Color], cubeMapCoord, 0);
+  }
+
+  if (flags_ & Flag::DepthTexture) {
+    frameBuffer_.attachCubeMapTexture(
+        Mn::GL::Framebuffer::BufferAttachment::Depth,
+        *textures_[TextureType::Depth], cubeMapCoord, 0);
+  } else {
+    frameBuffer_.attachRenderbuffer(
+        Mn::GL::Framebuffer::BufferAttachment::Depth, optionalDepthBuffer_);
+  }
+
+  mapForDraw();
+
+  frameBuffer_.clearDepth(1.0f).clearColor(0,                // color attachment
+                                           Mn::Vector4ui{0}  // clear color
+  );
+
+  CORRADE_INTERNAL_ASSERT(
+      frameBuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+      Mn::GL::Framebuffer::Status::Complete);
+}
+
+void CubeMap::mapForDraw() {
+  frameBuffer_.mapForDraw({
+      {Mn::Shaders::Generic3D::ColorOutput, colorAttachment},
+      // TODO:
+      //{Mn::Shaders::Generic3D::ObjectIdOutput, objectIdAttachment}
+  });
+}
+
+Mn::GL::CubeMapTexture& CubeMap::getTexture(TextureType type) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::getTexture():");
+  return *textures_[type];
+}
+
+#ifndef MAGNUM_TARGET_WEBGL
+// because Mn::Image2D image = textures_[type]->image(...)
+// requires desktop OpenGL
+bool CubeMap::saveTexture(TextureType type,
+                          const std::string& imageFilePrefix) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::saveTexture():");
+
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImageConverter> manager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImageConverter> converter;
+  if (!(converter = manager.loadAndInstantiate("AnyImageConverter"))) {
+    return false;
+  }
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    Mn::Image2D image = textures_[type]->image(
+        convertFaceIndexToCubeMapCoordinate(iFace), 0, {getPixelFormat(type)});
+
+    std::string filename = "";
+    switch (type) {
+      case TextureType::Color: {
+        filename = Cr::Utility::formatString("{}.{}.{}.png", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+      } break;
+
+      case TextureType::Depth: {
+        filename = Cr::Utility::formatString("{}.{}.{}.hdr", imageFilePrefix,
+                                             getTextureTypeFilenameString(type),
+                                             coordStrings[iFace]);
+      } break;
+    }
+    CORRADE_ASSERT(!filename.empty(),
+                   "CubeMap::saveTexture(): Unknown texture type.", false);
+
+    if (!converter->exportToFile(image, filename)) {
```

Comment:
For the depth texture, whose pixel format is R32F, would this (`converter->exportToFile(image, filename)`) be good enough to save the data as .hdr?

Reply:
Yes, the plugin will [internally expand to RGB, [as the docs say](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1StbImageConverter.html).